ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vrfs.py

```diff
@@ -28,17 +28,19 @@ def vrfs(self) -> dict | None:
         if not self._network_services_l3:
             return None
 
-        vrfs = {}
+        vrfs = []
+        new_vrf = {}
         for tenant in self._filtered_tenants:
             for vrf in tenant["vrfs"]:
                 vrf_name = vrf["name"]
                 if vrf_name == "default":
                     continue
 
-                if vrf_name in vrfs:
-                    self._raise_duplicate_vrf_error(vrf_name, tenant["name"], vrfs[vrf_name])
+                if new_vrf and vrf_name in new_vrf["name"]:
```

Comment:
I don't think this works if the duplicate VRFs are not directly adjacent.
You could store a list of vrf names and check on that. And then when you raise, you can pass `get_item(vrfs, "name", vrf_name)` as then third argument. `get_item` can be imported from `ansible_collections.arista.avd.plugins.plugin_utils.utils`.

Reply:
fixed